File: clients/android/sdk/src/main/kotlin/com/clienttools/sdk/inspector/ImageFileStore.kt
package com.clienttools.sdk.inspector

import android.content.Context
import android.util.Log
import java.io.File
import java.text.SimpleDateFormat
import java.util.*

class ImageFileStore(context: Context) {
    private val cacheDir = File(context.cacheDir, "inspector-images")
    private val TAG = "ImageFileStore"

    init {
        cacheDir.mkdirs()
    }

    fun saveImage(tag: String, timestamp: String, bytes: ByteArray, ext: String): ImageInfo? = try {
        val tagDir = File(cacheDir, tag).also { it.mkdirs() }
        val file = File(tagDir, "${tag}_${timestamp}.${ext}")
        file.writeBytes(bytes)
        ImageInfo(tag = tag, timestamp = timestamp, filePath = file.absolutePath, ext = ext)
    } catch (e: Exception) {
        Log.e(TAG, "Error saving image", e)
        null
    }

    fun getAllImages(): List<ImageInfo> = try {
        val result = mutableListOf<ImageInfo>()
        cacheDir.listFiles()?.forEach { tagDir ->
            if (!tagDir.isDirectory) return@forEach
            tagDir.listFiles()?.forEach { file ->
                val ext = file.extension.lowercase()
                if (ext != "png" && ext != "jpg" && ext != "jpeg") return@forEach
                val timestamp = parseTimestamp(file.name) ?: return@forEach
                result.add(ImageInfo(tagDir.name, timestamp, file.absolutePath, ext))
            }
        }
        result.sortedByDescending { it.timestamp }
    } catch (e: Exception) {
        Log.e(TAG, "Error listing images", e)
        emptyList()
    }

    fun getFilePath(tag: String, timestamp: String): String? {
        val tagDir = File(cacheDir, tag)
        return tagDir.listFiles()
            ?.firstOrNull { f ->
                val ext = f.extension.lowercase()
                (ext == "png" || ext == "jpg" || ext == "jpeg") && parseTimestamp(f.name) == timestamp
            }
            ?.absolutePath
    }

    fun deleteAll(): Boolean = try {
        cacheDir.deleteRecursively()
        cacheDir.mkdirs()
        true
    } catch (e: Exception) {
        Log.e(TAG, "Error deleting all", e)
        false
    }

    fun generateTimestamp(): String =
        SimpleDateFormat("MMdd-HHmm", Locale.US).format(Date())

    private fun parseTimestamp(filename: String): String? =
        Regex("""_(\d{4}-\d{4})\.\w+$""").find(filename)?.groupValues?.get(1)
}

File: clients/ios/sdk/Sources/Inspector/ImageFileStore.swift
import Foundation

class ImageFileStore {

    private let fileManager = FileManager.default
    private lazy var baseDir: URL = {
        let docs = fileManager.urls(for: .documentDirectory, in: .userDomainMask)[0]
        return docs.appendingPathComponent("client_tools/images")
    }()

    init() {
        try? fileManager.createDirectory(at: baseDir, withIntermediateDirectories: true)
    }

    func saveImage(tag: String, timestamp: String, bytes: Data, ext: String) -> ImageInfo? {
        let tagDir = baseDir.appendingPathComponent(tag)
        do {
            try fileManager.createDirectory(at: tagDir, withIntermediateDirectories: true)
            let fileURL = tagDir.appendingPathComponent("\(tag)_\(timestamp).\(ext)")
            try bytes.write(to: fileURL)
            return ImageInfo(tag: tag, timestamp: timestamp, filePath: fileURL.path, ext: ext)
        } catch {
            print("[ImageFileStore] saveImage error: \(error)")
            return nil
        }
    }

    func getAllImages() -> [ImageInfo] {
        var result: [ImageInfo] = []
        guard let tagDirs = try? fileManager.contentsOfDirectory(at: baseDir, includingPropertiesForKeys: [.isDirectoryKey]) else { return [] }
        for tagDir in tagDirs {
            guard (try? tagDir.resourceValues(forKeys: [.isDirectoryKey]))?.isDirectory == true else { continue }
            guard let files = try? fileManager.contentsOfDirectory(at: tagDir, includingPropertiesForKeys: nil) else { continue }
            for file in files {
                let ext = file.pathExtension.lowercased()
                guard ext == "png" || ext == "jpg" || ext == "jpeg" else { continue }
                guard let timestamp = parseTimestamp(file.lastPathComponent) else { continue }
                result.append(ImageInfo(tag: tagDir.lastPathComponent, timestamp: timestamp, filePath: file.path, ext: ext))
            }
        }
        return result.sorted { $0.timestamp > $1.timestamp }
    }

    func getFilePath(tag: String, timestamp: String) -> String? {
        let tagDir = baseDir.appendingPathComponent(tag)
        guard let files = try? fileManager.contentsOfDirectory(at: tagDir, includingPropertiesForKeys: nil) else { return nil }
        return files.first { f in
            let ext = f.pathExtension.lowercased()
            return (ext == "png" || ext == "jpg" || ext == "jpeg") && parseTimestamp(f.lastPathComponent) == timestamp
        }?.path
    }

    func deleteAll() -> Bool {
        do {
            try fileManager.removeItem(at: baseDir)
            try fileManager.createDirectory(at: baseDir, withIntermediateDirectories: true)
            return true
        } catch {
            print("[ImageFileStore] deleteAll error: \(error)")
            return false
        }
    }

    func generateTimestamp() -> String {
        let formatter = DateFormatter()
        formatter.dateFormat = "MMdd-HHmm"
        return formatter.string(from: Date())
    }

    private func parseTimestamp(_ filename: String) -> String? {
        let pattern = #"_(\d{4}-\d{4})\.\w+$"#
        guard let regex = try? NSRegularExpression(pattern: pattern),
              let match = regex.firstMatch(in: filename, range: NSRange(filename.startIndex..., in: filename)),
              let range = Range(match.range(at: 1), in: filename) else { return nil }
        return String(filename[range])
    }
}
